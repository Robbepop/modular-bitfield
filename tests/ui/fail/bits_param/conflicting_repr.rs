use modular_bitfield::prelude::*;

#[bitfield(bits = 16)]
#[repr(u32)]
pub struct SignInteger {
    sign: bool,
    value: B31,
}

fn main() {}
