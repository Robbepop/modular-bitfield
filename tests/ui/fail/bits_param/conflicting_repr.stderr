error: encountered conflicting `bits = 16` and #[repr(u32)] parameters
 --> tests/ui/fail/bits_param/conflicting_repr.rs:3:1
  |
3 | #[bitfield(bits = 16)]
  | ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `bitfield` (in Nightly builds, run with -Z macro-backtrace for more info)

error: conflicting `bits = 16` here
 --> tests/ui/fail/bits_param/conflicting_repr.rs:3:12
  |
3 | #[bitfield(bits = 16)]
  |            ^^^^

error: conflicting #[repr(u32)] here
 --> tests/ui/fail/bits_param/conflicting_repr.rs:4:8
  |
4 | #[repr(u32)]
  |        ^^^
